Modules/timemodule.c

```diff
@@ -2044,6 +2048,23 @@ PyInit_time(void)
     return PyModuleDef_Init(&timemodule);
 }
 
+#ifndef MS_WINDOWS
+static int
+sleep_unix(struct timeval timeout)
+{
+    int ret = 0;
+#ifdef HAVE_CLOCK_NANOSLEEP
+    struct timespec request;
+    request.tv_sec = timeout.tv_sec;
+    request.tv_nsec = (long)(timeout.tv_usec) * 1000;
+    ret = clock_nanosleep(CLOCK_MONOTONIC, 0, &request, NULL);
```

Comment:
I would prefer to avoid computing the relative 'timeout' value, but just pass 'deadline' absolute time since it's already computed: use flags=TIMER_ABSTIME.